pkg/storage/chunk/storage/caching_index_client_test.go

```diff
@@ -45,22 +45,25 @@ func TestCachingStorageClientBasic(t *testing.T) {
 	require.NoError(t, err)
 	logger := log.NewNopLogger()
 	cache := cache.NewFifoCache("test", cache.FifoCacheConfig{MaxSizeItems: 10, Validity: 10 * time.Second}, nil, logger)
-	client := newCachingIndexClient(store, cache, 1*time.Second, limits, logger, false)
+	client := newCachingIndexClient(store, cache, 1*time.Second, limits, logger, false, 10, 100)
 	queries := []chunk.IndexQuery{{
 		TableName: "table",
 		HashValue: "baz",
 	}}
 	err = client.QueryPages(ctx, queries, func(_ chunk.IndexQuery, _ chunk.ReadBatch) bool {
 		return true
 	})
+
 	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Instead of putting a sleep, I think you should check for the channel length to be 0 like:
```
	assert.Eventually(t, func() bool {
		return len(client.asyncQueue) == 0
	}, time.Second, 10*time.Millisecond)
```
It would check if the async queue got flushed every `10ms` for `1s`. If the queue length doesn't get to 0 in `1s` then the test would fail.

Reply:
> assert.Eventually(t, func() bool {
> 		return len(client.asyncQueue) == 0
> 	}, time.Second, 10*time.Millisecond)

ok,cool，👍

Reply:
done，thanks